api/v1alpha1/datadogagent_types.go

```diff
@@ -1145,11 +1151,18 @@ type DatadogAgentSpecClusterChecksRunnerSpec struct {
 // +k8s:openapi-gen=true
 type ImageConfig struct {
 	// Define the image to use:
-	// Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 6
+	// Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7
 	// Use "datadog/dogstatsd:latest" for Standalone Datadog Agent DogStatsD6
 	// Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent
+	// Use "agent" with the registry and tag configurations for <registry>/agent:<tag>
+	// Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>
 	Name string `json:"name"`
 
+	// Define the image version to use:
+	// To be used if the Name field doesn't correspond to a full image string.
+	// +optional
+	Tag string `json:"tag"`
+
```

Comment:
Open question: should we also provide the possibility to set if the user want to use the JMX image separately.

Something like:

```suggestion

	JMXSupportEnabled bool `json:"jmxSupportEnabled,omitempty"`
```

`if JMXSupportEnabled == true` the operator append to the end of the tag `-jmx`, and so the operator can still default properly the tag (7.28.0 for example)

Reply:
agreed, this can be useful!